python_modules/dagster/dagster/_core/execution/retries.py

```diff
@@ -82,7 +78,9 @@ def snapshot_attempts(self) -> Mapping[str, int]:
         return dict(self._attempts)
 
 
-def auto_reexecution_should_retry_run(instance: "DagsterInstance", run: DagsterRun):
+def auto_reexecution_should_retry_run(
+    instance: "DagsterInstance", run: DagsterRun, is_step_failure: bool
```

Comment:
i think you could make this the failure reason (Optional[RunFailureReason]) rather than a boolean? Pretty similar in practice, just move the is_step_failure check a bit lower